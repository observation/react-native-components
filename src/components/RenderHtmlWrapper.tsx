import React from 'react'
import { Dimensions, GestureResponderEvent, Text, View } from 'react-native'

import RenderHtml, { CustomRendererProps, defaultSystemFonts, RenderHTMLProps, TBlock } from 'react-native-render-html'

import ContentImage from './ContentImage'
import Log from '../lib/Log'
import { openUrl } from '../lib/Url'
import htmlStyle from '../styles/html'
import textStyle from '../styles/text'
import theme from '../styles/theme'

const imgRenderer = ({ tnode }: { tnode: TBlock }) => {
  const { src, alt } = tnode.attributes
  Log.debug('RenderHtmlWrapper:imgRenderer', src, alt)
  if (!src) {
    return null
  }
  return <ContentImage key={src} alt={alt} src={src} />
}

const olRenderer = ({ InternalRenderer, ...props }: CustomRendererProps<TBlock>) => (
  <InternalRenderer
    {...props}
    style={{
      ...props.style,
      paddingLeft: theme.margin.large,
    }}
  />
)

const ulRenderer = ({ TNodeChildrenRenderer, ...props }: CustomRendererProps<TBlock>) => (
  <>
    {props.tnode.children.map((item, i) => (
      <View key={i} style={{ flexDirection: 'row' }}>
        <Text style={{ marginRight: theme.margin.quarter }}>â€¢</Text>
        <TNodeChildrenRenderer tnode={item} />
      </View>
    ))}
  </>
)

const renderers = {
  img: imgRenderer,
  ol: olRenderer,
  ul: ulRenderer,
}

const RenderHtmlWrapper = ({
  contentWidth = Dimensions.get('window').width - 2 * theme.margin.common,
  renderersProps = {
    a: {
      onPress: (_event: GestureResponderEvent, href: string) => openUrl(href),
    },
  },
  baseStyle = textStyle.body,
  ...props
}: RenderHTMLProps) => {
  const systemFonts = [...defaultSystemFonts, 'Ubuntu']

  return (
    <RenderHtml
      contentWidth={contentWidth}
      renderersProps={renderersProps}
      renderers={renderers}
      systemFonts={systemFonts}
      baseStyle={baseStyle}
      tagsStyles={htmlStyle}
      enableExperimentalMarginCollapsing
      {...props}
    />
  )
}

export default RenderHtmlWrapper
