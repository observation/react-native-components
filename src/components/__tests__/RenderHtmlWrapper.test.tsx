import React from 'react'

import { render } from '@testing-library/react-native'

import RenderHtmlWrapper from '../RenderHtmlWrapper'

describe('RenderHtmlWrapper', () => {
  describe('Rendering', () => {
    describe('Custom image renderer', () => {
      test('Normal render', () => {
        const html = `<img src="https://waarneming.nl/image.jpeg" alt="A picture" />`
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })

      test('Without url', () => {
        const html = `<img alt="A picture" />`
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })
    })

    describe('Custom ordered list renderer', () => {
      test('Normal renderer', () => {
        const html = '<ol>\n<li>Ein</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ol>'
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })

      test('Ordered list inside an unorderer list', () => {
        const html =
          '<ul>\n<li>Ein<ol>\n<li>Ein</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ol>\n</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ul>'
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })

      test('Unorderer list inside an ordered list', () => {
        const html =
          '<ol>\n<li>Ein<ul>\n<li>Ein</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ul>\n</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ol>'
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })
    })

    describe('Custom unordered list', () => {
      test('Normal render', () => {
        const html = '<ul>\n<li>Ein</li>\n<li>Zwei</li>\n<li>Drei</li>\n</ul>'
        const { toJSON } = render(<RenderHtmlWrapper source={{ html }} />)
        expect(toJSON()).toMatchSnapshot()
      })
    })
  })
})
