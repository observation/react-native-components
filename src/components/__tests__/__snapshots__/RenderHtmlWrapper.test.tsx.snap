// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderHtmlWrapper Rendering Custom image renderer Normal render 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        {
          "margin": -16,
          "marginBottom": -8,
          "shadowColor": "#E6E6E6",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 4,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="photo"
      >
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#F8F9FA",
              "borderRadius": 8,
              "borderWidth": 1,
              "elevation": 6,
              "flexDirection": "row",
              "margin": 16,
              "overflow": "hidden",
              "shadowColor": "#939393",
            }
          }
        >
          <View
            style={
              {
                "borderRadius": 8,
                "margin": 16,
                "marginRight": 8,
                "overflow": "hidden",
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://waarneming.nl/image.jpeg",
                }
              }
              style={
                {
                  "height": 80,
                  "width": 80,
                }
              }
            />
          </View>
          <View
            style={
              {
                "flex": 1,
                "justifyContent": "center",
                "marginLeft": 8,
                "marginRight": 16,
              }
            }
          >
            <View>
              <Text
                numberOfLines={2}
                style={
                  {
                    "color": "#212121",
                    "fontFamily": "Ubuntu",
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "bold",
                    "lineHeight": 16,
                    "marginBottom": 4,
                  }
                }
              >
                A picture
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RenderHtmlWrapper Rendering Custom image renderer Without url 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  />
</View>
`;

exports[`RenderHtmlWrapper Rendering Custom ordered list renderer Normal renderer 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        [
          [
            {
              "marginBottom": 8,
              "marginTop": 16,
              "paddingLeft": 24,
            },
            null,
            null,
            {
              "paddingLeft": 0,
              "position": "relative",
            },
          ],
          undefined,
          undefined,
        ]
      }
      testID="ol"
    >
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            1. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Ein
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            2. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Zwei
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            3. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Drei
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RenderHtmlWrapper Rendering Custom ordered list renderer Ordered list inside an unorderer list 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Ein
      </Text>
      <View
        style={
          [
            [
              {
                "marginBottom": 8,
                "marginTop": 16,
                "paddingLeft": 24,
              },
              null,
              {
                "marginBottom": 0,
              },
              {
                "paddingLeft": 0,
                "position": "relative",
              },
            ],
            undefined,
            undefined,
          ]
        }
        testID="ol"
      >
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              undefined,
            ]
          }
          testID="marked-list-item"
        >
          <View
            style={
              [
                undefined,
                {
                  "width": 24,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="clip"
              numberOfLines={1}
              style={
                {
                  "alignSelf": "flex-end",
                  "color": "#666666",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontFamily": "Ubuntu",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "textAlign": "right",
                }
              }
              testID="marker-box"
            >
              1. 
            </Text>
          </View>
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 8,
                  },
                  undefined,
                  undefined,
                ]
              }
              testID="li"
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Ein
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              undefined,
            ]
          }
          testID="marked-list-item"
        >
          <View
            style={
              [
                undefined,
                {
                  "width": 24,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="clip"
              numberOfLines={1}
              style={
                {
                  "alignSelf": "flex-end",
                  "color": "#666666",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontFamily": "Ubuntu",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "textAlign": "right",
                }
              }
              testID="marker-box"
            >
              2. 
            </Text>
          </View>
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 8,
                  },
                  undefined,
                  undefined,
                ]
              }
              testID="li"
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Zwei
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "flexDirection": "row",
                "flexWrap": "nowrap",
              },
              undefined,
            ]
          }
          testID="marked-list-item"
        >
          <View
            style={
              [
                undefined,
                {
                  "width": 24,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="clip"
              numberOfLines={1}
              style={
                {
                  "alignSelf": "flex-end",
                  "color": "#666666",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontFamily": "Ubuntu",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                  "lineHeight": 20,
                  "textAlign": "right",
                }
              }
              testID="marker-box"
            >
              3. 
            </Text>
          </View>
          <View
            style={
              {
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 8,
                  },
                  undefined,
                  undefined,
                ]
              }
              testID="li"
            >
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Drei
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Zwei
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Drei
      </Text>
    </View>
  </View>
</View>
`;

exports[`RenderHtmlWrapper Rendering Custom ordered list renderer Unorderer list inside an ordered list 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        [
          [
            {
              "marginBottom": 8,
              "marginTop": 16,
              "paddingLeft": 24,
            },
            null,
            null,
            {
              "paddingLeft": 0,
              "position": "relative",
            },
          ],
          undefined,
          undefined,
        ]
      }
      testID="ol"
    >
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            1. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Ein
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginRight": 4,
                  }
                }
              >
                •
              </Text>
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Ein
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginRight": 4,
                  }
                }
              >
                •
              </Text>
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Zwei
              </Text>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "marginRight": 4,
                  }
                }
              >
                •
              </Text>
              <Text
                allowFontScaling={true}
                selectable={false}
                style={
                  [
                    {
                      "color": "#666666",
                      "fontFamily": "Ubuntu",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Drei
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            2. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Zwei
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignSelf": "stretch",
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            undefined,
          ]
        }
        testID="marked-list-item"
      >
        <View
          style={
            [
              undefined,
              {
                "width": 24,
              },
            ]
          }
        >
          <Text
            ellipsizeMode="clip"
            numberOfLines={1}
            style={
              {
                "alignSelf": "flex-end",
                "color": "#666666",
                "flexGrow": 0,
                "flexShrink": 0,
                "fontFamily": "Ubuntu",
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "normal",
                "lineHeight": 20,
                "textAlign": "right",
              }
            }
            testID="marker-box"
          >
            3. 
          </Text>
        </View>
        <View
          style={
            {
              "flexShrink": 1,
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 8,
                },
                undefined,
                undefined,
              ]
            }
            testID="li"
          >
            <Text
              allowFontScaling={true}
              selectable={false}
              style={
                [
                  {
                    "color": "#666666",
                    "fontFamily": "Ubuntu",
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                    "lineHeight": 20,
                  },
                  undefined,
                  undefined,
                ]
              }
            >
              Drei
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RenderHtmlWrapper Rendering Custom unordered list Normal render 1`] = `
<View
  style={
    [
      {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      [
        {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Ein
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Zwei
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "marginRight": 4,
          }
        }
      >
        •
      </Text>
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          [
            {
              "color": "#666666",
              "fontFamily": "Ubuntu",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
            },
            undefined,
            undefined,
          ]
        }
      >
        Drei
      </Text>
    </View>
  </View>
</View>
`;
